packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -452,22 +452,24 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
   }
 }
 
-function visitSha1s(object: any, sha1s: Set<string>) {
-  if (Array.isArray(object)) {
-    object.forEach(o => visitSha1s(o, sha1s));
-    return;
-  }
+function visitTraceEvent(object: any, sha1s: Set<string>): any {
+  if (Array.isArray(object))
+    return object.map(o => visitTraceEvent(o, sha1s));
+  if (object instanceof Buffer)
```

Comment:
Should we also skip it in the parent instead of inserting undefined?

Reply:
Luckily, `JSON.stringify()` does it for us.